### DISCUSSION TITLE
Completion menu & signature hover overlap.

### DISCUSSION THREAD

**Discussion by sitedyno** [folke/noice.nvim#600] [Q&A]
Is there a way to prevent this? The LSP is offering suggestions for named parameters which covers up the signature help.

![image](https://github.com/folke/noice.nvim/assets/216721/6ff6fbf8-ef43-47d8-95db-179620a83ad2)

Thank you 🙇‍♂️

**Comment by zer09** [comment]
If you have `hrsh7th/cmp-nvim-lsp-signature-help` or `lsp-saga` try to remove them.

https://github.com/folke/noice.nvim/issues/624#issue-1948875756

**Comment by sieuwerts** [comment]
I am having the same issue, but I'm not (to my knowledge) explicitly using any plugin other than `noice.nvim` for rendering the signature help. I'm using `lsp-zero` and my guess would be that this has something to do with it. It does a lot of things under the hood, and maybe it has to do with the way the `vim.lsp.handlers` for `textDocument/hover` is set up for both plugins that doesn't play well together.

I have tried my best to disable any interference from `lsp-zero` regarding this, but I'm not entirely sure that's what causing the issue. Because I can update the `lsp_doc_border` preset for `noice` and the border changes are applied, which leads me to believe that it actually _is_ `noice` that's in change of rendering it.

But it's merely my best guess at the moment, I will continue to investigate to see if I can find anything.

**Comment by sitedyno** [comment]
This is how I've worked around this for now using a custom view in noice:
``` lua
    lsp = {
        signature = {
            view = 'custom_signature',
        },
    },
    views = {
        custom_signature = {
            view = 'hover',
            anchor = 'SW',
            position = {
                row = 0,
            },
        },
    },
```
This results in the signature appearing above the cursor like so:
![image](https://github.com/folke/noice.nvim/assets/216721/32313f93-1c53-4aa5-be98-78233a160380)

**Reply by sieuwerts** [reply]
Thank you. I have tried this as well, the downside that I found of forcing the hover to be above the current line like this is if it's triggered when the cursor is close to the top of the screen. Then the hover might cover the current line, which is why I decided against it.

I did remove `lsp-zero` and also disabled hover in the `noice` config like so:
```
      hover = {
        enabled = false,
      },
      signature = {
        enabled = false,
        auto_open = {
          enabled = false,
          trigger = false,
        },
      },
```

But it still pops up automatically when in insert mode, so now I'm at a loss 😅  But I will continue to investigate what's triggering it, if there is another plugin somewhere that's gone rogue.
